src/Microsoft.OData.Core/UriParser/BuiltInUriFunctions.cs

```diff
@@ -535,5 +536,41 @@ private static FunctionSignatureWithReturnType[] CreateMathFunctionSignatureArra
 
             return new FunctionSignatureWithReturnType[] { doubleSignature, decimalSignature, nullableDoubleSignature, nullableDecimalSignature };
         }
+
+        private static void CreateLogicFunctions(IDictionary<string, FunctionSignatureWithReturnType[]> functions)
+        {
+            functions.Add("iif", CreateLogicFunctionSignatureArray());
+        }
+
+        private static FunctionSignatureWithReturnType[] CreateLogicFunctionSignatureArray()
+        {
+            var primitiveTypeKinds = Enum.GetValues(typeof(EdmPrimitiveTypeKind)).Cast<EdmPrimitiveTypeKind>().Where(k => k != EdmPrimitiveTypeKind.None && k != EdmPrimitiveTypeKind.Stream);
+            var result = new List<FunctionSignatureWithReturnType>();
+            foreach (var kind in primitiveTypeKinds)
+            {
+                var argumentType = GetPrimitiveTypeReference(kind, false);
+                result.Add(new FunctionSignatureWithReturnType(
+                    argumentType,
+                    EdmCoreModel.Instance.GetBoolean(false),
+                    argumentType,
+                    argumentType));

```

Comment:
How this is going to work when two argument differ just by the nullability. For example, if the second argument is string and second is nullable string. Your function could use the nullable version as the output should this happen.

Reply:
If one if nullable and another is not then both will be considered nullable. And function will match to nullable, nullable signature